pagopa/io-app
# chore: [IOAPPCOM-6] Fix due date visibility on `DueDateBar` component.

## Short description
This PR fixes the due date visibility inside the due date banner on messages.

In the current production release, messages with a set `due_date` have a box header presenting it that differs according to an expired or a valid/expiring due date.

| **Expired**  | **Valid/Expiring** |
| ------------- | ------------- |
|  <img src="https://user-images.githubusercontent.com/16268789/203299009-00b791fc-2460-4c6b-a2b8-ed6c34164512.png" width="200" /> | <img src="https://user-images.githubusercontent.com/16268789/203298981-b873a82c-f246-4014-840a-716897a98948.png" width="200" /> |

This PR aligns these two boxes presenting the same information for every due date state.

## List of changes proposed in this pull request

- Aligned the due date box between **expired** and **valid/expiring** states by **always showing both date and time** as follow:

| **Expired**  | **Expiring/Valid** |
| - | - | 
| <img src="https://user-images.githubusercontent.com/16268789/203531858-58ecaeca-290a-46e3-aaf2-e49dc45d375e.png" width="200" />  | <img src="https://user-images.githubusercontent.com/16268789/203301802-b318a419-ee12-4770-ab76-657235c5f107.png" width="200" /> |

<details>

<summary>Android/iOS Full Message Detail Screen Exmaples</summary>

| &nbsp; | **Expired**  | **Expiring/Valid** |
| - | - | - | 
| Android | <img src="https://user-images.githubusercontent.com/16268789/203302642-44cb6276-f15c-465a-8701-4a831fc68895.png" width="200" />  | <img src="https://user-images.githubusercontent.com/16268789/203302664-b7a0086d-2111-48b1-a378-b471fd90dbf8.png" width="200" /> |
| iOS | <img src="https://user-images.githubusercontent.com/16268789/203532070-216ccc0e-7ac9-4405-8347-d0c46f5269bd.png" width="200" />  | <img src="https://user-images.githubusercontent.com/16268789/203532076-7adc5871-fa6c-462a-8f5a-fbe12b80f53f.png" width="200" /> |
</details>

- Updated text labels and their translations with the new ones proposed by Product Team.
- Added test for "it" and "en" Locales on `DueDate.test.tsx` to test if the UI component renders as expected when it renders messages with different due dates. 

## How to test

- Run the application on both Android and iOS and check if messages with valid/expiring/expired states are rendered according to the above screenshots. 

- Check that the **ISO8601** due date string taken from the backend is rendered on the UI Component as a local date string with the date and time having the proper _hour shift_ given by the device's current set **Locale**.

- Run the `Render the DueDateBar accordingly with the due date` test inside `DueDate.test.tsx`.

<details>

<summary>How this test works</summary>

The test `Render the DueDateBar accordingly with the due date` searches the expected localized **date** and **time** string inside the rendered component by:<br/>
    - Using a set of test dates inside a mock data array made of the "date" and "time" component strings used to build an ISO8601 formatted string due dates (to forge the mock messages) and their corresponding _IT (Europe/Rome)_ and _EN (America/New_York)_ Locale hour shifts.
    - Converting the UTC (all dates inside jest have UTC Locale by a global setting) due date inside the mock message according to the hour offset of the selected Locale taken from the mock data. **Note:** Because both IT and EN chosen Locales use **daylight saving** times, the hour shift belongs to the actual date. It's not the same for every date, that's why the hour shift is inside the mock data array.
    - Render the UI Component with these mock message. 
    - Create runtime date and time local strings from the UTC date for _IT (Europe/Rome)_ and _EN (America/New_York)_ Locales to find them inside the `DueDateBar` component.
    - Check if these strings are actually present. 
    - Update test snapshots to freeze UI with the new layout and localized labels.
  
![due-date-test](https://user-images.githubusercontent.com/16268789/203524535-60f9a203-0793-4925-92c7-46d4db8549e4.png)

</details>
